### PR TITLE
[WIP] Enhance trashbin expiration settings

### PR DESCRIPTION
**Pull request by VicDeo**
According to https://github.com/owncloud/core/issues/16300

TODO:
- [ ] Migrate existing settings

### CODE REVIEW

**Comment by VicDeo** [comment]
@MTRichards can you proof-read a description in config.php?
https://github.com/owncloud/core/blob/trashbin-retention/config/config.sample.php#L402-L419

File: apps/files_trashbin/tests/expiration.php [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+use \OCA\Files_Trashbin\Expiration;
+require_once __DIR__.'/../lib/expiration.php';
```
Comment (DeepDiver1975) [inline apps/files_trashbin/tests/expiration.php]: the autoloaded should require the file - no need to explicit require it

Comment (VicDeo) [inline apps/files_trashbin/tests/expiration.php]: @DeepDiver1975 I use it to run  `phpunit expiration.php` for now. (And it hasn't passed the tests successfully yet)
The main question for me is  `did I understand correct how new values for this setting should work`.

**Comment by MTRichards** [comment]
How does this sound?

If the trash bin app is enabled (default), this setting defines the policy for when files and folders in the trash bin will be permanently deleted. The app allows for two settings, a minimum time for trash bin retention, and a maximum time for trash bin retention. Minimum time is the number of days a file will be kept, after which it may be deleted. Maximum time is the number of days at which it is guaranteed to be deleted. Both minimum and maximum times can be set together to explicitly define file and folder deletion. For migration purposes, this setting is installed initially set to "auto", which is equivalent to the default setting in ownCloud 8.1 and before. 

Available values:
- ``auto``          default setting. keeps files and folders in the trash bin for 30 days and automatically deletes anytime after that if space is needed (note: files may not be deleted if space is not needed).
- ``D, auto``     keeps files and folders in the trash bin for D+ days, delete anytime if space needed (note: files may not be deleted if space is not needed)
- ``auto, D``     delete all files in the trash bin that are older than D days automatically, delete other files anytime if space needed
- ``D1, D2``      keep files and folders the in trash bin for at least D1 days and delete when exceeds D2 days
- ``disabled``    trash bin auto clean disabled, files and folders will be kept forever

**Comment by VicDeo** [comment]
mmm. what the hell is that
>Caused by: hudson.plugins.git.GitException: Command "git checkout -f 9f44298255ff8ead8226d5c387b53a971f7a0fbc" returned status code 128:
18:19:42 stdout: 
18:19:42 stderr: fatal: reference is not a tree: 9f44298255ff8ead8226d5c387b53a971f7a0fbc

**Comment by DeepDiver1975** [comment]
> mmm. what the hell is that

That's github/jenkins fun - the branch is dead from Jenkins point of view. 
Since you are still developing - best is to create a new branch based on this one and submit a new pr.
To avoid such situations again:
- dont force push
- dont push while jenkins is running on a pr

**Pull request closed without merging**